xclim/indices/_multivariate.py

```diff
@@ -376,21 +383,59 @@ def heat_wave_frequency(
     characterize the occurrence of hot weather events that can result in adverse health outcomes for Canadian
     communities (Casati et al., 2013).
 
-    In Robinson (2001), the parameters would be `thresh_tasmin=27.22, thresh_tasmax=39.44, window=2` (81F, 103F).
+    The "consecutive" mode comes from Robinson (2001), where the parameters would be `thresh_tasmin=27.22, thresh_tasmax=39.44, window=2` (81F, 103F).
+
+    The "mean" mode comes from Lebel, Bustinza & Dubé (2017).
 
     References
     ----------
     Casati, B., A. Yagouti, and D. Chaumont, 2013: Regional Climate Projections of Extreme Heat Events in Nine Pilot
     Canadian Communities for Public Health Planning. J. Appl. Meteor. Climatol., 52, 2669–2698,
     https://doi.org/10.1175/JAMC-D-12-0341.1
 
+    Lebel, G., Bustinza, R. & Dubé, M. (2017). Analyse des impacts des vagues régionales de chaleur extrême sur la santé
+    au Québec de 2010 à 2015 : changements climatiques. Québec: INSPQ, Institut national de santé publique du Québec.
+    https://www.inspq.qc.ca/publications/2221
+    
     Robinson, P.J., 2001: On the Definition of a Heat Wave. J. Appl. Meteor., 40, 762–775,
     https://doi.org/10.1175/1520-0450(2001)040<0762:OTDOAH>2.0.CO;2
     """
+    if mode == "consecutive" and not np.isscalar(window):
+        raise ValueError(
+            "Window as sequence of weights is only valid for the 'mean' mode."
+        )
+
     thresh_tasmax = utils.convert_units_to(thresh_tasmax, tasmax)
     thresh_tasmin = utils.convert_units_to(thresh_tasmin, tasmin)
 
-    cond = (tasmin > thresh_tasmin) & (tasmax > thresh_tasmax)
+    if mode == "consecutive":
+        cond = (tasmin > thresh_tasmin) & (tasmax > thresh_tasmax)
+    elif mode == "mean":
+        if not np.isscalar(window):
+            weights = np.array(window)
+            window = len(window)
+            if weights.sum() != 1:
+                raise ValueError("Weights used for the window must sum to 1.")
+        else:
+            weights = np.array([1 / window] * window)
+        cond = (
+            _rolling(
+                tasmin,
+                window=window,
+                dim="time",
+                mode=lambda x, axis: np.sum(x * weights, axis=axis),
+            )
+            > thresh_tasmin
+        ) & (
+            _rolling(
+                tasmax,
+                window=window,
+                dim="time",
+                mode=lambda x, axis: np.sum(x * weights, axis=axis),
+            )
+            > thresh_tasmax
+        )
+
     group = cond.resample(time=freq)
     return group.apply(rl.windowed_run_events, window=window, dim="time")
```

Comment:
Line 440 - Would using ```mode='mean'``` not result in wanting to set window = 1 in the final ```rl.windowed_run_events``` call?  

I.e. if mode=='mean' and window == 3 we will already have done the rolling weighted average of n=3 so we do not need to find run_events of >= 3 as well in the final return right?